src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IProjectSnapshot.cs

```diff
@@ -21,6 +21,11 @@ internal interface IProjectSnapshot
     /// </summary>
     string FilePath { get; }
 
+    /// <summary>
+    /// Gets the full path to the folder under 'obj' where the project.razor.json file will live
+    /// </summary>
+    string IntermediateOutputPath { get; }
```

Comment:
Thats a good question, and I did think about putting both in the key, even if just for debugging purposes.

Removing them from the snapshot is a bigger question, and the reason for keeping them here now is that keys are created from snapshots, and not the other way around. That makes my life a lot easier at the moment, by reducing the places where keys are created. I am open to changing that in future if it makes sense. In general I'm open to changing most of this in future because the relationships between these classes is a little odd at times. Trying to keep that sort of thing out of these PRs to keep them managable.